swarm/storage/ldbstore.go

```diff
@@ -1049,7 +1049,9 @@ func (s *LDBStore) Close() {
 	s.lock.Unlock()
 	// force writing out current batch
 	s.writeCurrentBatch()
+	s.lock.Lock()
```

Comment:
why need?

Reply:
It is strange, but this is the data race: https://github.com/ethersphere/go-ethereum/issues/1198#issuecomment-461775049.